### DISCUSSION TITLE
Switch from files to database

### DISCUSSION THREAD

**Discussion by AlUlkesh** [AlUlkesh/stable-diffusion-webui-images-browser#22] [Announcements]
I'll commit another major change soon. I changed all the file-based handling to a database. This should make it easier for me to implement new features in the future.

On the first run, the old settings are migrated and some rescanning is necessary on each directory. Those are all one-time things though.

While not everything I want with this change is in there yet, everything should feel faster already.

Since this was a big rewrite, it's possible some data constellation might have a problem with the database implementation.

In case you run into a game-breaking bug, you can downgrade to the last non-db version by doing:
```
cd to extension-directory
git checkout 8bc7d3a11b4187d53905464327ea729f5c99f243
```
or to go back after that:
```
git checkout master
```

**Comment by AlUlkesh** [comment]
https://github.com/AlUlkesh/stable-diffusion-webui-images-browser/commit/44ce20fd595ffa47e82795b08d854529d12f5dfb

**Comment by Woisek** [comment]
Cool ... getting curious about where this is going to ... :)

**Comment by EllangoK** [comment]
I don't know the best place to ask this, but what do you think about using switching to SQLAlchemy instead of sqlite3. Its an ORM model so its a lot more pythonic, and we don't have to write any raw sequel. I think it should also make adding new features easier.

**Reply by AlUlkesh** [reply]
I'm not familiar with SQLAlchemy or ORMs. The reasons I chose sqlite3 are:
- It's packaged with python. I have some users who have setups that don't let them install a simple package like send2trash correctly.
- I'm more familiar with SQL than anything else in this project.

So... sorry, I don't think I will switch at this point.

**Reply by EllangoK** [reply]
fair enough, no worries. And yeah I was thinking about that first part as it easier since its included by default.
